### PR TITLE
Fix an error in float not equal operation

### PR DESCRIPTION
**Pull request by gergocs**
Signed-off-by: Gergo Csizi gergocs@inf.u-szeged.hu

### CODE REVIEW

**Comment by zherczeg** [comment]
Please add some tests

**Comment by gergocs** [comment]
Patch is ready for review

**Comment by zherczeg** [comment]
How do you detected this issue? Normally, a not equal should fail if either is nan, but it will return with true here.

File: test/jit/f64-operations.wast [hunk 1]
```diff
@@ -161,6 +161,12 @@
 
     select
 )
+
+(func (export "test28") (result i32)
+    f64.const -0x0p+0
+    f64.const -nan
+    f64.ne
```
Comment (zherczeg) [inline test/jit/f64-operations.wast:168]: You could add more nan checks with multiple results.

**Comment by gergocs** [comment]
The [reference manual](https://github.com/sunfishcode/wasm-reference-manual/blob/master/WebAssembly.md#floating-point-inequality) says the not equal operation should be `unordered or not equal`.

File: test/jit/f64-operations.wast [hunk 2]
```diff
@@ -180,3 +198,5 @@
 (assert_return (invoke "test25") (i32.const 0) (i32.const 1) (i32.const 0))
 (assert_return (invoke "test26") (f64.const 6.6))
 (assert_return (invoke "test27") (i32.const 6))
+(assert_return (invoke "test28") (i32.const 1) (i32.const 1) (i32.const 0) (i32.const 1))
+
```
Comment (zherczeg) [inline test/jit/f64-operations.wast:202]: No need this extra newline

**Review by zherczeg — APPROVED** [review]
LGTM

**Pull request merged**